src/annotator/integrations/vitalsource.ts

```diff
@@ -198,7 +198,13 @@ export class VitalSourceInjector {
       }
       contentFrames.add(frame);
       onDocumentReady(frame, (err, document_) => {
-        const body = document_?.body;
+        if (err) {
+          return;
+        }
+
+        // If `err` is null, then `document_` will be set.
+        const body = document_!.body;
```

Comment:
This function uses a callback with `(error, document)` arguments, where only one is set, as that's a conventional thing to do in JavaScript. It's not as friendly for TS as making the callback type `(errorOrDocment: Error|Document) => void` though. I might revisit this later.

Reply:
Yeah, I think we should be sparing with our `!` suffixes, as they sort of mean we're operating without TS guardrails in reality.